client/src/components/App.js

```diff
@@ -5,22 +5,71 @@ import NavHeader from "./NavHeader";
 import Creator from "./Creator";
 import Example from "./Example";
 import Fan from "./Fan";
+import Web3 from "web3";
+import Web3Modal from "web3modal";
+import Fortmatic from "fortmatic";
+
+let provider = null;
+let web3Modal = null;
+let web3 = null;
+let accounts = null;
+let mainAddress = "";
+
+const providerOptions = {
+  fortmatic: {
+    package: Fortmatic, // required
+    options: {
+      key: process.env.FORTMATIC_KEY, // required
+    },
+  },
+};
 
 class App extends Component {
   constructor(props) {
     super(props);
+    this.state = {
+      connectedWallet: "",
+    };
     this.openWalletConnectModal = this.openWalletConnectModal.bind(this);
   }
 
-  openWalletConnectModal() {
-    // TODO: Open web3modal and set connectedWallet when rendering Fan.
-    console.log("Opened wallet connect");
+  async openWalletConnectModal() {
+    if (!provider) {
+      web3Modal = new Web3Modal({
+        cacheProvider: true, // optional
+        providerOptions, // required
+      });
+      provider = await web3Modal.connect();
+      web3 = new Web3(provider);
+    }
+    if (!accounts) {
+      accounts = await web3.eth.getAccounts();
+      mainAddress = accounts[0].toLowerCase();
```

Comment:
Do we need to lowercase? I thought casing usually matters for the ETH addresses, but maybe the address verifier is smart enough to check it.

If this was for stylistic purposes in the navheader, Uniswap keeps upper/lower case: 
![Screen Shot 2021-01-26 at 11 29 02 PM](https://user-images.githubusercontent.com/2449384/105957995-de57f300-602e-11eb-8e7a-a2a483059500.png)